model_compression_toolkit/target_platform_capabilities/schema/v1.py

```diff
@@ -361,91 +431,182 @@ class OperatorsSetBase(TargetPlatformModelComponent):
+    name: Optional[str] = None  # Will be set in the validator
+    type: Literal["OperatorSetConcat"] = "OperatorSetConcat"
+
+    model_config = ConfigDict(frozen=True)
+
+    @model_validator(mode='before')
+    def validate_and_set_name(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Validate the input and set the concatenated name based on the operators_set.
+
+        Args:
+            values (Dict[str, Any]): Input data.
+
+        Returns:
+            Dict[str, Any]: Modified input data with 'name' set.
+        """
+        operators_set = values.get('operators_set', ())
 
-    def __post_init__(self):
+        if values.get('name') is None:
+            # Generate the concatenated name from the operator sets
+            concatenated_name = "_".join([
+                op.name.value if isinstance(op.name, OperatorSetNames) else op.name
+                for op in operators_set
+            ])
+            values['name'] = concatenated_name
+
+        # Ensure qc_options is None
+        values['qc_options'] = None
+
+        return values
+
+    @model_validator(mode='after')
+    def validate_after_initialization(cls, op_set_concat: 'OperatorSetConcat') -> 'TargetPlatformModel':
+        """
+        Perform validation after the model has been instantiated.
+
+        Args:
+            op_set_concat (OperatorSetConcat): The instantiated fusing.
+
+        Returns:
+            OperatorSetConcat: The validated model.
+        """
+        if not op_set_concat.operators_set:
+            Logger.critical("OperatorSetConcat requires at least one OperatorsSet in 'operators_set'.") # pragma: no cover
+
+        if op_set_concat.type != 'OperatorSetConcat':
+            Logger.critical("'type' field must not change from default value 'OperatorSetConcat'.")# pragma: no cover
+
+        return op_set_concat
+
+    def get_info(self) -> Dict[str, Any]:
         """
-        Post-initialization processing to generate the concatenated name and set it as the `name` attribute.
+        Get information about the concatenated operator sets as a dictionary.
 
-        Calls the parent class's __post_init__ method and creates a concatenated name
-        by joining the names of all operator sets in `operators_set`.
+        Returns:
+            Dict[str, Any]: A dictionary containing the concatenated name.
         """
-        # Generate the concatenated name from the operator sets
-        concatenated_name = "_".join([op.name.value if hasattr(op.name, "value") else op.name for op in self.operators_set])
-        # Set the inherited name attribute using `object.__setattr__` since the dataclass is frozen
-        object.__setattr__(self, "name", concatenated_name)
+        return {"name": self.name, "operators_set": [op.get_info() for op in self.operators_set]}
 
 
-@dataclass(frozen=True)
 class Fusing(TargetPlatformModelComponent):
```

Comment:
so what was it in operatorset about fusion?

Reply:
I don't understand the question